src/compiler/crystal/codegen/fun.cr

```diff
@@ -466,6 +474,7 @@ class Crystal::CodeGenVisitor
     return if arg.name == "_"
 
     target_def_var = target_def_vars.try &.[arg.name]
+    location = target_def_var.try &.location || target_def.location
```

Comment:
```suggestion
    location = target_def_var.try(&.location) || target_def.location
```

Reply:
If you saw my comment: it is work in progress. So I committed it for other to try but final refactoring and cleanup is still in progress.

Reply:
Oh, I missed that! 👍 Should I remove the review comments for now?

Reply:
No, keep it. It will remind me what needs to be cleaned up later on.